dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/api/imports/HarvestedJsonParserTest.java

```diff
@@ -1,359 +0,0 @@
-package edu.harvard.iq.dataverse.api.imports;
-
-import edu.harvard.iq.dataverse.DatasetFieldServiceBean;
-import edu.harvard.iq.dataverse.persistence.dataset.ControlledVocabularyValue;
-import edu.harvard.iq.dataverse.persistence.dataset.DatasetFieldType;
-import edu.harvard.iq.dataverse.util.json.JsonParseException;
-import org.junit.jupiter.api.*;
-import org.junit.jupiter.api.extension.*;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.Mockito;
-import org.mockito.junit.jupiter.MockitoExtension;
-import org.mockito.junit.jupiter.MockitoSettings;
-import org.mockito.quality.Strictness;
-
-import javax.json.Json;
-import javax.json.JsonObject;
-import javax.json.JsonReader;
-import java.io.StringReader;
-import java.util.Arrays;
-import java.util.List;
-import java.util.stream.Collectors;
-
-import static org.junit.jupiter.api.Assertions.*;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyBoolean;
-import static org.mockito.ArgumentMatchers.anyString;
-
-@ExtendWith(MockitoExtension.class)
-@MockitoSettings(strictness = Strictness.LENIENT)
-public class HarvestedJsonParserTest {
```

Comment:
I'm kind of suprised that we are removing this unit test in favour of IT. As it would be the easiest solution to test some uncommon scenarios when parsing dataset fields (we are not bounded to the current db state - where it could be a little harder to find fields that have desired multiplicity/whether is compound or not/or is it have vocabulary or not). Not an issue - but it seems that we are pointing to direction that will be harder to test.

Reply:
This is something what I've mentioned on a daily meeting. These unit tests were testing just `parseVocabulary` fields, so the method was left `public` just to be able to test it like that. 
Leaving it `private` and unit testing its parts (like `parseVocabulary`) by mocking whole Dataset is IMO not readable. 
I'm leaving it to your suggestion - leave the method `public` and keep unit tests or mock whole Dataset keep it private?

Reply:
Ok. Let's leave IT way.